src/backend/commands/explain.c

```diff
@@ -724,6 +724,9 @@ ExplainOnePlan(PlannedStmt *plannedstmt, IntoClause *into, ExplainState *es,
 		/* run the plan */
 		ExecutorRun(queryDesc, dir, 0L, true);
 
+		/* run cleanup too */
+		ExecutorFinish(queryDesc);
```

Comment:
Although the results are the same, I'm not sure that we need move this line to L745 ?

Reply:
I synced with @fanfuxiaoran yesterday :
GPDB cherry-pick breaks the change we made at https://github.com/apache/cloudberry/pull/622/files#diff-5f24db464bbd016410e96159d19e5eddaaeaf016905ba4efc8fbac982a95574e which resolved the issue explain(analyze) hang.